litmus-portal/backend/subscriber/pkg/cluster/cluster.go

```diff
@@ -2,88 +2,173 @@ package cluster
+	"github.com/litmuschaos/litmus/litmus-portal/backend/subscriber/pkg/k8s"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
-	memory "k8s.io/client-go/discovery/cached"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/restmapper"
 	"log"
+
+	memory "k8s.io/client-go/discovery/cached"
 )
 
-var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+const (
+	PortalConfigName = "litmus-portal-config"
+	DefaultNamespace = "litmus"
+)
 
-func applyRequest(obj *unstructured.Unstructured, requestType interface{}, dr dynamic.ResourceInterface) (*unstructured.Unstructured, error) {
+var (
+	Ctx             = context.Background()
+	decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	dr              dynamic.ResourceInterface
+)
 
-	if requestType == "create" {
-		response, err := dr.Create(context.TODO(), obj, metav1.CreateOptions{})
+// IsClusterConfirmed checks if the config map with "is_cluster_confirmed" is true or not.
+func IsClusterConfirmed(clusterData map[string]string) (bool, error) {
+	clientset, err := k8s.GetGenericK8sClient()
+	if err != nil {
+		return false, err
+	}
+
+	configMapData := map[string]string{
+		"is_cluster_confirmed": "false",
+	}
+
+	configMap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PortalConfigName,
+		},
+		Data: configMapData,
+	}
+
+	getCM, err := clientset.CoreV1().ConfigMaps(DefaultNamespace).Get(Ctx, PortalConfigName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, err := clientset.CoreV1().ConfigMaps(DefaultNamespace).Create(Ctx, &configMap, metav1.CreateOptions{})
 		if err != nil {
-			return nil, fmt.Errorf("err: %v\n", err)
+			return false, err
+		}
+	} else {
+		if getCM.Data["is_cluster_confirmed"] == "true" {
+			clusterData["KEY"] = getCM.Data["cluster_key"]
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// ClusterRegister function creates litmus-portal config map in the litmus namespace
+func ClusterRegister(clusterData map[string]string) (bool, error) {
+	clientset, err := k8s.GetGenericK8sClient()
+	if err != nil {
+		return false, err
+	}
+
+	configMapData := map[string]string{
+		"is_cluster_confirmed": "true",
+		"cluster_key":          clusterData["KEY"],
+		"cluster_id":           clusterData["CID"],
+	}
+
+	newConfigMap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PortalConfigName,
+		},
+		Data: configMapData,
+	}
+
+	_, err = clientset.CoreV1().ConfigMaps(DefaultNamespace).Update(Ctx, &newConfigMap, metav1.UpdateOptions{})
```

Comment:
Wouldn't it be better to create the configmap in this function rather than creating a dummy one first in the `isClusterConfirmed` function?

Reply:
Yes, it is possible. Thanks for finding this

Reply:
Moved to clusterRegister()